xen/arch/arm/include/asm/smccc.h

```diff
@@ -265,6 +266,45 @@ void __arm_smccc_1_0_smc(register_t a0, register_t a1, register_t a2,
         else                                                    \
             arm_smccc_1_0_smc(__VA_ARGS__);                     \
     } while ( 0 )
+
+/**
+ * struct arm_smccc_1_2_regs - Arguments for or Results from SMC call
+ * @a0-a17 argument values from registers 0 to 17
+ */
+struct arm_smccc_1_2_regs {
+    unsigned long a0;
+    unsigned long a1;
+    unsigned long a2;
+    unsigned long a3;
+    unsigned long a4;
+    unsigned long a5;
+    unsigned long a6;
```

Comment:
We should use a fixed size type here. I am wondering wether we should use uint64_t or register_t

Reply:
This was reviewed at https://lore.kernel.org/all/C8575668-5771-4E77-B77C-15E6E740E540@arm.com/

I'll keep it as it is for now and let you comment on the mailing list.

Reply:
agree after looking at the rest it is already done this way.